LoRaEngine/LoraKeysManagerFacade/IoTCentralImp/DeviceTwinPageResult.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoraKeysManagerFacade.IoTCentralImp
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+    using LoraKeysManagerFacade.IoTCentralImp.Definitions;
+    using Newtonsoft.Json.Linq;
+
+    internal class DeviceTwinPageResult : IRegistryPageResult<IDeviceTwin>
+    {
+        private readonly HttpClient client;
+        private readonly string apiVersion;
+        private readonly Func<DeviceTemplateInfo, string> query;
+        private readonly IEnumerable<DeviceTemplateInfo> deviceTemplateInfos;
+        private readonly IEnumerator<DeviceTemplateInfo> deviceTemplatesEnumerator;
+
+        public DeviceTwinPageResult(HttpClient client, IEnumerable<DeviceTemplateInfo> deviceTemplateInfos, string apiVersion, Func<DeviceTemplateInfo, string> query)
+        {
+            this.client = client;
+            this.deviceTemplateInfos = deviceTemplateInfos;
+            this.apiVersion = apiVersion;
+            this.query = query;
+
+            deviceTemplatesEnumerator = deviceTemplateInfos.GetEnumerator();
+            HasMoreResults = deviceTemplatesEnumerator.MoveNext();
+        }
+
+        public bool HasMoreResults { get; private set; }
+
+        public async Task<IEnumerable<IDeviceTwin>> GetNextPageAsync()
+        {
+            var results = new List<IDeviceTwin>();
+
+            if (!this.HasMoreResults)
```

Comment:
Property access doesn't need a `this` qualifier:

```suggestion
            if (!HasMoreResults)
```